src/rpp/rpp/operators/reduce.hpp

```diff
@@ -55,10 +55,11 @@ struct reduce_t : public operators::details::operator_observable_strategy<reduce
     using ResultValue = Seed;
 };
 
-template<rpp::constraint::decayed_type Seed, rpp::constraint::observer TObserver, rpp::constraint::decayed_type Accumulator>
+template<rpp::constraint::observer TObserver, rpp::constraint::decayed_type Accumulator>
 struct reduce_no_seed_observer_strategy
 {
     using DisposableStrategyToUseWithThis = rpp::details::none_disposable_strategy;
+    using Seed = std::invoke_result_t<Accumulator, utils::convertible_to_any, utils::convertible_to_any>;
```

Comment:
@victimsnino What do you think? Is it to remove Seed symbol from strategy template

Reply:
At this point Seed is rpp::utils::extract_observer_type_t<TObserver>

Reply:
But you proposal wouldn't work for templated lambdas

Reply:
Same about `reduce_observer_strategy` and same about scan operator..

Reply:
Indeed, let me revert

Reply:
Could you update to rpp::utils::extract_observer_type_t everywhere instead?

Reply:
done